diff -ur a/arch/x86/Makefile b/arch/x86/Makefile
--- a/arch/x86/Makefile	2021-02-14 22:32:24.000000000 +0000
+++ b/arch/x86/Makefile	2021-02-15 19:11:59.696317934 +0000
@@ -25,7 +25,7 @@
 # How to compile the 16-bit code.  Note we always compile for -march=i386;
 # that way we can complain to the user if the CPU is insufficient.
 REALMODE_CFLAGS	:= -m16 -g -Os -DDISABLE_BRANCH_PROFILING \
-		   -Wall -Wstrict-prototypes -march=i386 -mregparm=3 \
+		   -Wall -Wstrict-prototypes --target=i386 -mregparm=3 \
 		   -fno-strict-aliasing -fomit-frame-pointer -fno-pic \
 		   -mno-mmx -mno-sse $(call cc-option,-fcf-protection=none)
 
diff -ur a/arch/x86/boot/compressed/Makefile b/arch/x86/boot/compressed/Makefile
--- a/arch/x86/boot/compressed/Makefile	2021-02-14 22:32:24.000000000 +0000
+++ b/arch/x86/boot/compressed/Makefile	2021-02-15 19:11:41.632209528 +0000
@@ -33,7 +33,7 @@
 KBUILD_CFLAGS += -Wundef
 KBUILD_CFLAGS += -DDISABLE_BRANCH_PROFILING
 cflags-$(CONFIG_X86_32) := -march=i386
-cflags-$(CONFIG_X86_64) := -mcmodel=small -mno-red-zone
+cflags-$(CONFIG_X86_64) := -mcmodel=small -mno-red-zone --target=x86_64-pc-linux-gnu
 KBUILD_CFLAGS += $(cflags-y)
 KBUILD_CFLAGS += -mno-mmx -mno-sse
 KBUILD_CFLAGS += -ffreestanding -fshort-wchar
diff -ur a/fs/fuse/virtio_fs.c b/fs/fuse/virtio_fs.c
--- a/fs/fuse/virtio_fs.c	2021-10-13 07:42:04.000000000 +0000
+++ b/fs/fuse/virtio_fs.c	2021-10-16 19:05:33.843931165 +0000
@@ -1447,7 +1447,7 @@
 	fc->release = fuse_free_conn;
 	fc->delete_stale = true;
 	fc->auto_submounts = true;
-	fc->sync_fs = true;
+	fc->sync_fs = false;
 
 	/* Tell FUSE to split requests that exceed the virtqueue's size */
 	fc->max_pages_limit = min_t(unsigned int, fc->max_pages_limit,
