diff -ur a/fs/fuse/virtio_fs.c b/fs/fuse/virtio_fs.c
--- a/fs/fuse/virtio_fs.c	2021-10-13 07:42:04.000000000 +0000
+++ b/fs/fuse/virtio_fs.c	2021-10-16 19:05:33.843931165 +0000
@@ -1447,7 +1447,7 @@
 	fc->release = fuse_free_conn;
 	fc->delete_stale = true;
 	fc->auto_submounts = true;
-	fc->sync_fs = true;
+	fc->sync_fs = false;
 
 	/* Tell FUSE to split requests that exceed the virtqueue's size */
 	fc->max_pages_limit = min_t(unsigned int, fc->max_pages_limit,
